ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/StreamInsertsSubscriber.java

```diff
@@ -46,10 +47,10 @@ protected void afterSubscribe(final Subscription subscription) {
   }
 
   @Override
-  protected void handleValue(final KsqlObject row) {
+  protected void handleValue(final Object row) {
     httpRequest.writeCustomFrame(
         0, 0,
-        Buffer.buffer().appendString(row.toJsonString()).appendString("\n")
+        Buffer.buffer().appendString(JsonObject.mapFrom(row).toString()).appendString("\n")
```

Comment:
Thanks @survivant . This makes sense to me, though I don't think it works to make this change directly in StreamInsertsSubscriber, since `JsonObject.mapFrom(row).toString()` produces a different result from `row.toJsonString()` in the case where `row` is a `KsqlObject`. The former results in
```
{"empty":false,"map":{"k":"v"}}
```
while the latter produces the desired
```
{"k":"v"}
```
It's too bad that Java won't allow us to support separate client methods for the two types of inserts (`KsqlObject` and generic POJO). I think the best we can do is have a check here to see if `row` is an instance of a `KsqlObject` and perform the conversion to a string differently if so. Not the prettiest but it should work.

Do you want to drive this PR to completion by adding test coverage and the relevant documentation updates?

Reply:
I see your point.  I can try do to another pass to make it cleaner.  I'll try my best to give testcases and documentation, but I'm involve in other opensource projects, so I don't have a lot of free time.